chainntnfs/btcdnotify/btcd.go

```diff
@@ -387,12 +387,13 @@ func (b *BtcdNotifier) attemptHistoricalDispatch(msg *confirmationsNotification,
 		h := t.TxSha()
 		if txHash.IsEqual(&h) {
 			txIndex = uint32(i)
+			break
 		}
 	}
 
 	confDetails := &chainntnfs.TxConfirmation{
 		BlockHash:   blockHash,
-		BlockHeight: uint32(currentHeight) - uint32(tx.Confirmations),
+		BlockHeight: uint32(currentHeight) - uint32(tx.Confirmations) + 1,
```

Comment:
Excellent catch, here my off-by-one senses were tingling when I originally wrote this fragment. Good to see the bug corrected. However, the same error was noticed by @bryanvu and fixed in his latest PR. If you rebase to the latest version of master, the part of the diff should disappear.